tools/VscPreviewExtension/src/childProcessHost.ts

```diff
@@ -74,24 +60,51 @@ export class ChildProcessHost {
         let docContent = doc.getText();
         let fileName = doc.fileName;
         let rootPath = workspace.rootPath;
-        let filePath;
+        let relativePath;
         if (!rootPath || !fileName.includes(rootPath)) {
-            let indexOfFilename = fileName.lastIndexOf("\\");
-            rootPath = fileName.substr(0, indexOfFilename);
-            filePath = fileName.substring(indexOfFilename + 1);
+            let indexOfFileName = fileName.lastIndexOf("\\");
+            rootPath = fileName.substr(0, indexOfFileName);
+            relativePath = fileName.substring(indexOfFileName + 1);
         } else {
             let rootPathLength = rootPath.length;
-            filePath = fileName.substr(rootPathLength + 1, fileName.length - rootPathLength);
+            relativePath = fileName.substr(rootPathLength + 1, fileName.length - rootPathLength);
         }
-        if (doc.languageId === ConstVariable.languageId) {
-            let numOfRow = doc.lineCount;
-            this.writeToStdin(rootPath, filePath, numOfRow, docContent);
+        if (doc.languageId === "markdown") {
+            this.sendHttpRequestCore(rootPath, relativePath, docContent);
         }
     }
 
-    protected writeToStdin(rootPath: string, filePath: string, numOfRow: number, docContent: string) { }
+    protected sendHttpRequestCore(rootPath: string, relativePath: string, docContent: string) {
+        window.showErrorMessage(`[Extension Error]: Not supported`);
+    }
 
-    protected appendWrap(content) {
-        return content + "\n";
+    protected newHttpServerAndStartPreview(activeTextEditor) {
+        window.showInformationMessage("Environment initializing, please wait several seconds!");
+        let that = this;
+        let http = require("http");
+        let server = http.createServer();
+        server.listen(0);
+        server.on('listening', function () {
+            var port = server.address().port;
+            server.close();
```

Comment:
>close [](start = 19, length = 5)

after the server is closed and set port of ChildProcessHost, could exit the function?